internal/libtor/enabled.go

```diff
@@ -0,0 +1,277 @@
+//go:build ooni_libtor
+
+package libtor
+
+// Adapted from https://github.com/cretz/bine
+// SPDX-License-Identifier: MIT
+
+//
+// #cgo linux,amd64 CFLAGS: -I${SRCDIR}/linux/amd64/include
+// #cgo linux,amd64 LDFLAGS: -L${SRCDIR}/linux/amd64/lib -ltor -levent -lssl -lcrypto -lz -lm
+//
+// #cgo android,arm CFLAGS: -I${SRCDIR}/android/arm/include
+// #cgo android,arm LDFLAGS: -L${SRCDIR}/android/arm/lib -ltor -levent -lssl -lcrypto -lz
+// #cgo android,arm64 CFLAGS: -I${SRCDIR}/android/arm64/include
+// #cgo android,arm64 LDFLAGS: -L${SRCDIR}/android/arm64/lib -ltor -levent -lssl -lcrypto -lz
+// #cgo android,386 CFLAGS: -I${SRCDIR}/android/386/include
+// #cgo android,386 LDFLAGS: -L${SRCDIR}/android/386/lib -ltor -levent -lssl -lcrypto -lz
+// #cgo android,amd64 CFLAGS: -I${SRCDIR}/android/amd64/include
+// #cgo android,amd64 LDFLAGS: -L${SRCDIR}/android/amd64/lib -ltor -levent -lssl -lcrypto -lz
+//
+// #include <limits.h>
+// #include <stdbool.h>
+// #include <stdlib.h>
+//
+// #include <tor_api.h>
+//
+// /* Note: we need to define inline helpers because we cannot index C arrays in Go. */
+//
+// static char **cstringArrayNew(size_t size) {
+//     char **argv = calloc(sizeof(char *), size);
```

Comment:
IIUC, `size` is the length of the string array, correct?
In this case, I think it should be `char **argv = calloc(size, sizeof(char *));` https://en.cppreference.com/w/c/memory/calloc
Or not?

Reply:
They're both `size_t`. The first one is `count` and the second one is `size`. The man page on macOS says: "allocates enough space for count objects that are size bytes of memory each". So, swapping the arguments as you suggest is definitely more consistent with the spirit of the lib call, good catch 🙌 💯 🔝. (In practical terms, the code was nonetheless working as intended because the two `size_t` values are multiplied (with overflow checks) to obtain the total amount of bytes to allocate.)